src/main/java/ai/chalk/internal/arrow/FeatherProcessor.java

```diff
@@ -321,19 +332,21 @@ public static byte[] inputsToArrowBytes(Map<String, List<?>> inputs) throws Exce
             }
         }
 
-        VectorSchemaRoot root = VectorSchemaRoot.of(Utils.listToArray(fieldVectors, FieldVector.class));
-
+        byte[] byteArr;
         try (
+            VectorSchemaRoot root = VectorSchemaRoot.of(Utils.listToArray(fieldVectors, FieldVector.class));
             ByteArrayOutputStream out = new ByteArrayOutputStream();
             ArrowFileWriter writer = new ArrowFileWriter(root, null, Channels.newChannel(out));
         ) {
             writer.start();
             writer.writeBatch();
             writer.end();
-            writer.close();
-            root.close();
-            return out.toByteArray();
+            byteArr = out.toByteArray();
+        }
+        for (RootAllocator allocator : allocators) {
+            allocator.close();
```

Comment:
would do this in a finally block and wrap the whole loop with a try

Reply:
addressed